### DISCUSSION TITLE
VisionOS support

### DISCUSSION THREAD

**Discussion by CrawlyKatt** [alvr-org/ALVR#1969] [General]
Even if needed to be side loaded and with zero controller support, Apple VisionOS support would enable ALVR to be the first PCVR solution for Vision Pro users. Users could use Playspace calibrator in order to use index controllers, or even quest pro controllers. Would it be possible to implement a basic version that is 6dof head tracking only? There is no rush, please dont trash the headset in the discussion. VisionOS is not the headset, visionOS is the platform.

**Comment by zmerp** [comment]
This is probably very hard, unless someone is willing to buy one and contribute the changes needed to make it work. The two main changes are the VR runtime (OpenXR vs visionOS SDK) and the decoder (MediaCodec vs Video Toolbox). As for the graphics API, this can be handled by the wgpu abstraction layer (coming soon).

**Reply by CrawlyKatt** [reply]
nice nice I'm glad to see your thoughts would need to be done. Im full of hope and wish I could contribute. I'm still a long ways away from owning a VisionOS device myself for now.

**Comment by illuminatedRealityStudios** [comment]
I would love to see this happen.. so far I love my Apple Vision Pro… - i mean you can use the xcode simulator too… 
until you can get a hold of one.

**Reply by zmerp** [reply]
Check out [alvr-visionos](https://github.com/alvr-org/alvr-visionos). In the span of a few weeks, some very talented people added full fledged support for the Apple Vision Pro, with controller support using Joycons.
